python/strings_udf/cpp/include/cudf/strings/udf/numeric.cuh

```diff
@@ -0,0 +1,74 @@
+
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "udf_string.cuh"
+
+#include <cudf/strings/detail/convert/int_to_string.cuh>
+#include <cudf/strings/detail/convert/string_to_float.cuh>
+#include <cudf/strings/detail/convert/string_to_int.cuh>
+
+namespace cudf {
+namespace strings {
+namespace udf {
+
+/**
+ * @brief Converts a string into an integer
+ *
+ * The '+' and '-' are allowed but only at the beginning of the string.
+ * The string is expected to contain base-10 [0-9] characters only.
+ * Any other character will end the parse.
+ * Overflow of the int64 type is not detected.
+ */
+__device__ inline int64_t stoi(string_view const& d_str)
+{
+  return cudf::strings::detail::string_to_integer(d_str);
+}
+
+/**
+ * @brief Converts an integer into string
+ *
+ * @param value integer value to convert
+ */
+__device__ inline udf_string to_string(int64_t value)
+{
+  udf_string result;
+  if (value == 0) {
+    result.append("0");
+    return result;
+  }
+  auto const d_value = static_cast<double>(abs(value));
+  auto digits        = static_cast<int32_t>(log10(d_value)) + 1 + (value < 0);
```

Comment:
I don't think this function is sufficient for values with large magnitudes. Here is a demonstration of several failing cases (which should be explicitly tested in this PR) and a proposed alternative method that solves these problems:

https://godbolt.org/z/6hdxKs8aM

Reply:
Thanks Bradley. Good catch.
This was meant to be a place holder until I could make `count_digits` available here.
I should've added a TODO to remind me to fix this.

Reply:
The [`count_digits` logic](https://github.com/rapidsai/cudf/blob/416d4d5d194bf42ffe7ca0c0090860f3c91f90aa/cpp/include/cudf/strings/detail/convert/int_to_string.cuh#L69-L93) looks good to me. Nice!

Reply:
With an unrollable loop? https://godbolt.org/z/oEWbbdjj6